apps/dashboard/app/(app)/logs/components/charts/hooks.ts

```diff
@@ -0,0 +1,65 @@
+import { trpc } from "@/lib/trpc/client";
+import type { LogsTimeseriesDataPoint } from "@unkey/clickhouse/src/logs";
+import { addMinutes, format } from "date-fns";
+import { useMemo } from "react";
+import { useLogSearchParams } from "../../query-state";
+import { type TimeseriesGranularity, getTimeseriesGranularity } from "../../utils";
+
+const roundToSecond = (timestamp: number) => Math.floor(timestamp / 1000) * 1000;
+
+const formatTimestamp = (value: string | number, granularity: TimeseriesGranularity) => {
+  const date = new Date(value);
+  const offset = new Date().getTimezoneOffset() * -1;
+  const localDate = addMinutes(date, offset);
+
+  switch (granularity) {
```

Comment:
is the granularity passed straight from query params?
I'd prefer if we do `"hour" | "day" | ...`, it's consistent with other pages and our API

Reply:
We calculate the distance between start and end then decide on which granularity to show